app/models/book.rb

```diff
@@ -3,6 +3,7 @@ class Book < ActiveRecord::Base
   has_many :order_items
 
   scope :by_published, -> { order(published_date: :desc) }
+  scope :most_popular, -> { all.includes(:order_items).sort_by(&:times_sold).reverse! }
```

Comment:
I was just talking to Annie about this too. I'll look into the counter caches you suggested. I hadn't seen them before, but it seems like it may make things easier.